src/components/AccessControlDialog.vue

```diff
@@ -0,0 +1,291 @@
+  components: {
+    GirderBreadcrumb,
+    GirderSearch,
+  },
+  inject: ['girderRest'],
+  props: {
+    value: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+    model: {
+      type: Object,
+      default: null,
+      validator: model =>
+        createLocationValidator(false)(model) && model._modelType !== 'user',
+    },
+  },
+  data() {
+    return {
+      public: false,
+      access: null,
+      recursive: false,
+    };
+  },
+  computed: {
+    groupsAndUsers() {
+      if (!this.access) {
+        return [];
+      }
+      return [...this.access.groups, ...this.access.users];
+    },
+    permissions() {
+      return [
+        {
+          text: 'Can view',
+          value: 0,
+        },
+        {
+          text: 'Can edit',
+          value: 1,
+        },
+        {
+          text: 'Is owner',
+          value: 2,
+        },
+      ];
+    },
+  },
+  watch: {
+    value(value) {
+      if (value) {
+        this.getAccessControlData();
+      } else {
+        this.access = null;
+      }
+    },
+  },
+  methods: {
+    async getAccessControlData() {
+      this.public = this.model.public;
+      const { data: access } = await this.girderRest.get(`${this.model._modelType}/${this.model._id}/access`);
+      this.access = access;
+    },
+    remove(model) {
+      let index = this.access.groups.indexOf(model);
+      if (index !== -1) {
+        this.access.groups.splice(index, 1);
+        return;
+      }
+      index = this.access.users.indexOf(model);
+      if (index !== -1) {
+        this.access.users.splice(index, 1);
+      }
+    },
+    groupOrUserSelected(selectedModel) {
+      if (!this.groupsAndUsers.find(model => model.id === selectedModel._id)) {
+        this.access[`${selectedModel._modelType}s`].push({
+          flags: [],
+          id: selectedModel._id,
+          level: 0,
+          name: selectedModel.login
+            ? `${selectedModel.firstName} ${selectedModel.lastName}`
+            : selectedModel.name,
+          [selectedModel.login ? 'login' : 'description']:
+            selectedModel.login || selectedModel.description,
+        });
+      }
+    },
+    async save() {
+      const access = {
+        users: this.access.users.map(({ id, level, flags }) => ({
+          id,
+          level,
+          flags,
+        })),
+        groups: this.access.groups.map(({ id, level, flags }) => ({
+          id,
+          level,
+          flags,
+        })),
+      };
+      const data = {
+        access: JSON.stringify(access),
+        public: this.public,
+        publicFlags: [],
+        recurse: this.recursive,
+        progress: true,
+      };
+      await this.girderRest.put(
+        `${this.model._modelType}/${this.model._id}/access`,
+        stringify(data),
+      );
+      this.$emit('input', false);
+      this.$emit('model-access-changed', this.model);
+    },
+  },
+};
+</script>
+
+<template lang="pug">
+  v-dialog(
```

Comment:
So far I've avoided wrapping components in `v-dialog` and have put this burden on the consumer app.  I think it would be good to be consistent with `UpsertFolder` and  `Upload` and remove the dialog wrapper so that permissions could potentially be used in other ways.  I like the idea of this component taking only a `model`.

Reply:
A good question! The reason I put it in a dialog and even suffix the naming of the component Dialog is that I can't think of a situation a consumer of this component won't use this in a dialog. I have used `UpsertFolder` and `Upload` as standalone components (not with DataBrowser) in other projects, but they all have been put in dialogs. I personally believe GWC should provide easy to use components and common sense convention, but If we would use this not in a dialog, I am happy to extract the component but would still keep the dialog component for ease of use.

Reply:
I've used Upload standalone without a dialog.  Also, dialog comes with a lot of configuration that users may want to change (persistent, width, scroll, etc.) and we don't want to restrict that or forward it through our own props, so IMO it's best to let apps set up their dialog.

Reply:
I would say that hiding the ability to control these props enables me to provide easy to use common sense component. To be specific, for the width, I know the content won't look too good on a wide space so in this case, I decided for them (max-width 800px). For scrollable, because of the list of permission could go long, so I provided a scroll on the content. Also, for dialog scrollable to work, the content needs to have a v-card-text, which is hidden to the consumer unless open up the source code. 
But again, this is based on the assumption that the access control is better to be used in a dialog. 
The repo is not a general UI library, so I think should provide really easy to use components that simply works like that. 
This is the same reason I internalized `UpsertFolder` and `Upload` dialog to `DataBrowser`  https://github.com/girder/girder_web_components/pull/125. Because for the three times I use DataBrowser, I always wanted `UpsertFolder` and `Upload` to show just like they are in the demo app, but instead I needed to create the dialogs, hooking up the callbacks correctly, handling show state myself three times. By internalizing the dialog, the saved effort is huge. (See code reduction of demo/App.vue https://github.com/girder/girder_web_components/pull/125/files?file-filters%5B%5D=.vue)